src/main/java/ch/bfh/bti7081/presenter/MainViewPresenter.java

```diff
@@ -0,0 +1,48 @@
+package ch.bfh.bti7081.presenter;
+
+import ch.bfh.bti7081.model.manager.SeminarManager;
+import ch.bfh.bti7081.model.seminar.Seminar;
+import ch.bfh.bti7081.view.MainViewContent;
+import com.vaadin.flow.component.html.H2;
+import com.vaadin.flow.component.html.Label;
+import com.vaadin.flow.component.orderedlayout.VerticalLayout;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.*;
+import java.util.stream.Collectors;
+public class MainViewPresenter {
+    public MainViewPresenter(MainViewContent view) {
+        // generate next seminars on startup
+        view.setFeatureView(this.getNextSeminaries());
+    }
+
+    /**
+     * creates a VerticalLayout with a title and the next few seminaries to display on the homepage
+     * @return VerticalLayout to add to view
+     */
+    private VerticalLayout getNextSeminaries() {
```

Comment:
getNextSeminaries müsste m.E. statisch sein...
Zudem ist meinem Verständnis nach MVP nicht vollumfassend umgesetzt. 
Das GUI sollte m.E. im Presenter zwar informationstechnisch abgefüllt werden, aber nicht mit Vaadin-Objekten. Sonst macht die Unterteilung in View und Presenter nur wenig Sinn. Auch ist der Presenter im Moment mit dem Modell (Manager-Klasse) fest verdratet. Im Beispiel zu MVP wird die Manager-Klasse dem Konstruktor des Presenters übergeben.

Reply:
danke, passe ich an

Reply:
Ursprünglich hatte ich ein Manager-Objekt im Container und dies dem Presenter übergeben. Da die getSeminaries-Methode auf dem Manager statisch ist, hat IntelliJ empfohlen. dies über `SeminarManager.getSeminaries();` anzusprechen, was das Manager-Objekt überflüssig gemacht hat